### DISCUSSION TITLE
November 2023 Update Report

### DISCUSSION THREAD

**Discussion by MotoLegacy** [nzp-team/nzportable#498] [Announcements]
### Introduction
This post is also available on [ModDB](https://www.moddb.com/games/nazi-zombies-portable/news/november-2023-update-report).

November was a busy month! It started out rather slow but very quickly became packed with significant overhauls on every platform, and a lot of new content. There was so much here I could not mention because it would bloat this post up even more. Be sure to check out my [Patreon](https://www.patreon.com/CypressImplex) to continue supporting the project!

Making the headlines for this will be hard, but here's my best shot!

### Considerable AI Improvements

https://github.com/nzp-team/nzportable/assets/26030250/0d4cef13-f4c7-46e9-a33b-98e0c555d056

Rejoice! This month a lot of work has been put into improving the responsiveness of AI, as well as fixing edge cases where their pathfinding would fail! There is still work to be done in this regard, so keep an eye on the commit history to see this improve further. Inherited from the FTE builds, on other platforms AI will now decide to follow the player directly instead of following path nodes if certain thresholds are met, and these thresholds have been improved to avoid running into knee high obstacles or failing to turn corners!

Both Hellhounds and Crawlers now have considerable fixes too -- Hellhounds now "stalking" their prey before running at them at high speeds, and Crawler bounding boxes being much larger to avoid the player intersecting with them.

### Improvements to Gibbing, and general FTE parity

https://github.com/nzp-team/nzportable/assets/26030250/feda0cc3-6ccf-44db-bfcb-e1ec170ef841

Following up on improved AI behaviors, the FTE builds of the game have received more AI specific improvements, namely the restoration of bullet penetration (which was available on other platforms already) and their gibbing effects. Additionally -- gibbing on all platforms has been made more accurate with Call of Duty: World at War. It should be easier to trigger a headless bleed-out, or knocking the arms off of zombies.

### Progress on Co-operative Play

https://github.com/nzp-team/nzportable/assets/26030250/ef80da9b-f15e-4f4f-927b-7daa7370e220

[![Nazi Zombies: Portable PS Vita + Nintendo Switch Multiplayer Demonstration](http://img.youtube.com/vi/O1T5nCX4ANQ/0.jpg)](http://www.youtube.com/watch?v=O1T5nCX4ANQ "Nazi Zombies: Portable PS Vita + Nintendo Switch Multiplayer Demonstration")

A lot of work this month has been dedicated to Co-Op play, on all platforms. FTE now features a Cooperative play menu with allows users to more easily link-up and play. PlayStation VITA and Nintendo Switch builds of the game are now also able to connect to each other -- the start of proper cooperative play for all console versions of the game. This ability still isn't accessible via the front end as there are still issues to work out, though as of right now it is very promising.

Additionally, third person player animations now exist for most actions, and FTE now features a more streamlined chat ability. Effort has also been made to optimize network rate exchanges for a decrease in latency.

### Pack-A-Punch Rewrite

https://github.com/nzp-team/nzportable/assets/26030250/0f07e1b5-6093-4b69-b829-a6492df62015

The Pack-A-Punch machine has received a considerable overhaul, now being properly scaled (needs mapper opt-in), and featuring dynamism such as light emission, and a status flag!

### NEW MAP: "Warehouse" Remake

[!["Warehouse" - Nazi Zombies: Portable Trailer](http://img.youtube.com/vi/sd4cPQwDcF4/0.jpg)](http://www.youtube.com/watch?v=sd4cPQwDcF4 "\"Warehouse\" - Nazi Zombies: Portable Trailer")

Dug up from old archives of unused content from years ago, and then restored in full, "WAREHOUSE" returns as a brand new experience from the NZ:P Team officially, with secrets to discover and a story to be told.

### NEW MAP: "Warehouse (Classic)"

![fte-20231201005941-0](https://github.com/nzp-team/nzportable/assets/26030250/2c55e4ef-d795-48a0-8b43-4d6924d002f0)

Along with the remake distributed, the original "WAREHOUSE" was rebuilt as well, bringing it up to a higher standard as well as restoring content missing that was featured in it's original outing in 2011. Warehouse has been returned to it's former glory.

### NEW MAP: "Hangar" by DR_Tinklez

![fte-20231201140030-0](https://github.com/nzp-team/nzportable/assets/26030250/063d3cca-71e1-4d3a-bbfc-4afffb60a90e)

It's fight or flight and you can't find fuel. If only scavenging for fuel is as easy as erasing the horde.

### MAP UPDATE: "Christmas Special"

![fte-20231201133115-0](https://github.com/nzp-team/nzportable/assets/26030250/9f106c96-45f2-428c-855c-12536881e1dd)

Not left hanging -- "Christmas Special" also had minor updates to fix considerable issues and restore content missing from this re-release. This update also expands it's Mystery Box weapon list considerably, a hold over from past limitations.

### MAP UPDATE: "Bunker Defense"

![fte-20231201133318-0](https://github.com/nzp-team/nzportable/assets/26030250/604b9c81-0ab4-4c97-aabe-6eb0724da5ba)

Revnova's "Bunker Defense" has been updated, adding new areas and secrets, considerably improving this simple-yet-enjoyable challenge map.

### Carpenter Power-Up Overhaul

https://github.com/nzp-team/nzportable/assets/26030250/a6abacca-ee89-41f5-a888-5667a613e2bd

Potentially a minor detail for most -- Carpenter was a very unintelligent Power-Up. Simply repairing all of the barricades in a map at the same time, and acting as though all barriers were completely down before. This update brings more parity with the Power-Up -- it now cycles through every unrepaired barrier, and repairs the required boards needed (as opposed to assuming they were all already down).

### Player Stance-change Improvements

https://github.com/nzp-team/nzportable/assets/26030250/cca3056f-018c-4528-9d02-a3e59d741bbe

The behavior for changing stances has been improved and made more accurate -- Player's now have their bounding box adjusted when crouching, making them able to fit into smaller sized spaces. Going prone also now prohibits walking while aiming down the sights or firing your weapon, to counter-act the increase in accuracy.

### Misc. Bug Fix/Small Addition Bullet List

**FTE (PC):**

- Settings now properly apply when pressing "Save & Apply" on the main menu.
- Improved the Game Over UI screen.
- Menu scaling now properly applies based on window size.
- Player movement is now client-predicted.
- Increased the size of the circular crosshair used on the Wunderwaffe DG-2

**Global:**

- Increased the time between rounds to better match Call of Duty.
- Addressed an issue where headshots were dealing 1.5x the amount of expected damage.
- Fixed instances where automatic reload was not triggering.
- Addressed an issue where explosives could hurt AI through walls.
- Addressed an issue where explosives would damage other players.
- The player landing sound now plays less frequently.
- The player's "jump" sound now only triggers when performing a Dolphin Dive.
- Considerable teleporter improvements.
- Map makers can now opt to use Double-Tap 1.0 instead of being forced to use Double-Tap 2.0.

And more, and more, and more.. See you next month!

**Thanks to my [Patrons](https://www.patreon.com/CypressImplex)!**

- mochi
- Rinse a Gateau Mandate
- Benjamin Engelhardt
- botdog
- Derped Crusader
- Ryan Baldwin
- LEXX
- MissLavender

**Comment by Jein1** [comment]
Where i can find the link to download it? ☺️

**Reply by klimscha** [reply]
https://www.moddb.com/games/nazi-zombies-portable

**Reply by MotoLegacy** [reply]
https://github.com/nzp-team/nzportable/releases/latest

**Reply by MotoLegacy** [reply]
> https://www.moddb.com/games/nazi-zombies-portable

no... everything there is from 2014 and prior.

**Comment by Zdravko004** [comment]
Would you consider mobile port for Android any time in the future??
